tutorials/cosmodc2/cosmoDC2_TAP_access.md

```diff
@@ -0,0 +1,183 @@
+
++++
+
+## What is the default maximum number of rows returned by the service?
+
+This service will return a maximum of 2 billion rows by default.
+
+```{code-cell} ipython3
+service.maxrec
+```
+
+This default maximum can be changed, and there is no hard upper limit to what it can be changed to.
+
+```{code-cell} ipython3
+print(service.hardlimit)
+```
+
+## List the columns in the chosen table
+
+This table contains 301 columns.
+
+```{code-cell} ipython3
+columns = tables[tablename].columns
+print(len(columns))
+```
+
+Let's learn a bit more about them.
+
+```{code-cell} ipython3
+for col in columns:
+    print(f'{f"{col.name}":30s}  {col.description}')
+```
+
+## Create a histogram of redshifts
+
+Let's figure out what redshift range these galaxies cover. Since we found out above that it's a large catalog, we can start with a spatial search over a small area of 0.1 deg. The ADQL that is needed for the spatial constraint is:
+
+```{code-cell} ipython3
+adql = f"SELECT redshift FROM {tablename} WHERE CONTAINS(POINT('ICRS', RAMean, DecMean), CIRCLE('ICRS',54.218205903,-37.497959343,.1))=1"
+adql
+```
+
+Now we can use the previously-defined service to execute the query with the spatial contraint.
+
+```{code-cell} ipython3
+cone_results = service.search(adql)
+```
+
+```{code-cell} ipython3
+# Plot a histogram
+import numpy as np
+import matplotlib.mlab as mlab
+import matplotlib.pyplot as plt
+
+num_bins = 20
+# the histogram of the data
+n, bins, patches = plt.hist(cone_results['redshift'], num_bins, 
+                            facecolor='blue', alpha = 0.5)
+plt.xlabel('Redshift')
+plt.ylabel('Number')
+plt.title('Redshift Histogram CosmoDC2 Mock Catalog V1 abridged')
+```
+
+We can easily see form this plot that the simulated galaxies go out to z = 3.
+
++++
+
+## Visualize galaxy colors at z ~ 0.5
+
+Now let's visualize the galaxy main sequence at z = 2.0. First, we'll do a narrow redshift cut with no spatial constraint.
+
+Let's do it as an asynchronous search since this might take awhile.
+
+```{code-cell} ipython3
+service = vo.dal.TAPService("https://irsa.ipac.caltech.edu/TAP")
+adql = f"SELECT Mag_true_r_sdss_z0, Mag_true_g_sdss_z0, redshift FROM {tablename} WHERE redshift > 0.5 and redshift < 0.54"
+results = service.run_async(adql)
+```
+
+```{code-cell} ipython3
+len(results['mag_true_r_sdss_z0'])
+```
+
+```{code-cell} ipython3
+# Since this results in almost 4 million galaxies, 
+# we will construct a 2D histogram rather than a scatter plot.
+plt.hist2d(results['mag_true_r_sdss_z0'], results['mag_true_g_sdss_z0']-results['mag_true_r_sdss_z0'],
+           bins=200, cmap='plasma', cmax=500)
+
+# Plot a colorbar with label.
+cb = plt.colorbar()
+cb.set_label('Number')
+
+# Add title and labels to plot.
+plt.xlabel('SDSS Mag r')
+plt.ylabel('SDSS rest-frame g-r color')
+
+# Show the plot.
+plt.show()
+```
+
+## About this notebook
+
+ * Author: Vandana Desai (IRSA Science Lead)
```

Comment:
cc @vandesai1 to OK this. We assumed the [original notebook](https://irsa.ipac.caltech.edu/data/theory/Cosmosims/gator_docs/CosmoDC2_Mock_V1_Catalog.html) was Dave's, but he said you wrote it.